File: sentimental-credit/credit.py
card_number = input("Number: ")
length=len(card_number)
if 13<=length<=16:
    sum=0
    try:
        for i in range(-1,-length-1,-1):
            if (i%2==1):
                sum+=int(card_number[i])
            else:
                tem=int(card_number[i])*2
                if(tem//10==0):
                    sum+=tem
                else:
                    sum+=(tem%10+tem//10)
        if sum%10==0:
            if (card_number[0]=="3" and card_number[1] in "47" and length==15):
                print("American Express")
            elif (length==16 and card_number[0]=="5" and card_number[1] in "12345"):
                print("MASTERCARD")
            elif ((length==16 or length==13) and card_number[0]=="4"):
                print("VISA")
        else:
            print("INVALID")
    except ValueError:
        print("Wrong Entry")
else:
    print("INVALID")

File: C_basics/credit.c
#include <cs50.h>
#include <stdio.h>

long get_input();

int main(void)
{
    long card_no=0;
    do{
    card_no=get_input();
    }
    while (card_no<0);


    long last_digit=card_no;
    int n=0;
    int odd_sum=0,even_sum=0;

    //verification algorithm
    while(card_no!=0)
    {
    int new_sum=0;
    int temp=0;

        if(n%2==0)
        {
            odd_sum+=card_no%10;
        }
        else
        {
            temp=(card_no%10)*2;

            if(temp%10!=temp)
            {
                new_sum=temp%10+1;
            }
            else
            {
                new_sum=temp;
            }
            even_sum+=new_sum;
        }

        last_digit=card_no;
        card_no=(card_no-card_no%10)/10;
        n+=1;
    }


    int validity=(even_sum+odd_sum)%10;

    //size validity checker
    if (validity==0){
        printf("VALID\n");
    }
    else{
        printf("INVALID\n");
    }

    //trying classification of brands
    if (13<=n && n<=16)
    {
        if (last_digit==4)
        {
            printf("VISA\n");
        }
        else if (last_digit==3)
        {
            printf("AMERICAN EXPRESS\n");
        }
        else if (last_digit==5)
        {
            printf("MASTERCARD\n");
        }
        else{
            printf("INVALID\n");
        }
    }
    else{
        printf("INVALID\n");
    }
}

long get_input()
{
    long card_no=get_long("Enter your card number(13-16 digits)");
    return card_no;
}